test/utility.test.ts

```diff
@@ -0,0 +1,232 @@
+    },
+
+    result: true,
+  },
+  {
+    name: "should return true when valid element",
+    jsonXMLElementKey: "RequestedAttribute",
+    jsonXMLElementValue: {
+      "@isRequired": "true",
+      "@Name": "Lastname",
+      "@xmlns": "urn:oasis:names:tc:SAML:2.0:metadata",
+    },
+
+    result: true,
+  },
+];
+
+const samlExtensionElementSchemaNamespaceCheck: any[] = [
+  {
+    name: "should throw JSONXMLSchemaError error when missing element namespace",
+    value: {
+      "md:RequestedAttribute": {
+        "@isRequired": "true",
+        "@Name": "Lastname",
+      },
+    },
+    result: {
+      throw: {
+        message: "Namespace @xmlns:md is not defined for element md:RequestedAttribute",
+      },
+    },
+  },
+  {
+    name: "should throw JSONXMLSchemaError error when missing element namespace",
+    value: {
+      vetuma: {
+        LG: {
+          "#text": "sv",
+        },
+      },
+    },
+    result: {
+      throw: {
+        message: "Namespace @xmlns is not defined for element vetuma",
+      },
+    },
+  },
+  {
+    name: "should throw JSONXMLSchemaError error when wrong element namespace",
+    value: {
+      "md:RequestedAttribute": {
+        "@isRequired": "true",
+        "@Name": "Lastname",
+      },
+      vetuma: {
+        "@xmlns": "urn:vetuma:SAML:2.0:extensions",
+        LG: { "#text": "sv" },
+      },
+    },
+    result: {
+      throw: {
+        message: "Namespace @xmlns:md is not defined for element md:RequestedAttribute",
+      },
+    },
+  },
+  {
+    name: "should return true when valid element",
+    value: {
+      "md:RequestedAttribute": {
+        "@isRequired": "true",
+        "@Name": "Lastname",
+        "@xmlns:md": "urn:vetuma:SAML:2.0:extensions",
+      },
+    },
+    result: true,
+  },
+  {
+    name: "should return true when valid element",
+    value: {
+      RequestedAttribute: {
+        "@isRequired": "true",
+        "@Name": "Lastname",
+        "@xmlns": "urn:oasis:names:tc:SAML:2.0:metadata",
+      },
+      vetuma: {
+        "@xmlns": "urn:oasis:names:tc:SAML:2.0:metadata",
+        LG: {
+          "#text": "sv",
+        },
+      },
+    },
+    result: true,
+  },
+];
+
+describe("test validateXMLNamespace", () => {
+  it("should be exists", () => {
+    should.exist(validateXMLNamespace);
+  });
+
+  it("should be function", () => {
+    validateXMLNamespace.should.be.Function();
+  });
+
+  it("should throw TypeError when values are invalid", () => {
+    should(() => {
+      validateXMLNamespace("", undefined);
+    }).throw();
+    should(() => {
+      validateXMLNamespace("", undefined);
+    }).throw(TypeError);
+    should(() => {
+      validateXMLNamespace("", undefined);
+    }).throw("key should be define");
+    should(() => {
+      validateXMLNamespace("test", {});
+    }).throw("test XML Element value should be object and not empty");
+  });
+
+  xmlElementSchemaNamespaceCheck.forEach((check) => {
```

Comment:
I'm not a big fan of this pattern. I'd rather just see the outer "describe" block (XML Element Schema Namespace Check) with a inner "it" block with each test in the array. That makes it much easier to debug tests, and, in this case, would probably actually be fewer lines of code. Thoughts @markstos ?

Reply:
I agree. It would be more clear not use the big array and format each each test in the standard way.

Reply:
@cjbarth just double checking, You want to keep this "it" block array in another "describe" block, right?

Reply:
Based on the new simplified spec, we'll just need a test to make sure that the payload gets interested in the correct place in the resultant XML and that a bad data type causes and error when constructing. In that case, we don't have to worry about many of these tests; they can be removed.